src/requirements/reqs-functions.ts

```diff
@@ -257,23 +257,22 @@ export function computeRequirements(
     userChoiceOnDoubleCountingElimiation: [],
     // TODO assign an unique ID to each requirement entry.
     getRequirementUniqueID: requirement => `${requirement.name} ${requirement.description}`,
-    getCourseUniqueID: course => course.code,
+    getCourseUniqueID: course => `${course.roster} ${course.courseId}`,
     getAllCoursesThatCanPotentiallySatisfyRequirement: requirement => (
       requirement.courses
         .map((eligibleCourses): readonly CourseTaken[] => {
           const courses: CourseTaken[] = [];
-          Object.entries(eligibleCourses).forEach(([roster, mapping]) => {
-            Object.entries(mapping).forEach(([subject, numbers]) => {
-              numbers.forEach(number => {
-                courses.push({
-                  roster,
-                  subject,
-                  code: `${roster}: ${subject} ${number}`,
-                  number,
-                  credits: 0
-                });
-              });
-            });
+          Object.entries(eligibleCourses).forEach(([roster, courseIds]) => {
+            courseIds.forEach(courseId => courses.push({
+              roster,
+              courseId,
+              // Only roster and courseId are used for equality comparison,
+              // so other dummy values doesn't matter.
+              code: 'DUMMY',
```

Comment:
Is there a reason you are setting these to 'DUMMY' instead of just removing the parameters altogether from the typescript definition?

Reply:
It's still used for university req computation

https://github.com/cornell-dti/course-plan/blob/b87611aad3ab68d93136bb551900a8678e644c5f/src/requirements/reqs-functions.ts#L83-L84